File: 6-kyu/give-me-a-diamond/python/solution.py
func diamond(_ size: Int) -> String? {
  if size % 2 == 0 || size <= 0 { 
    return nil
  }    
  var ret = ""
  var spacesBefore : Int = size / 2
  for n in stride(from: 1, to: size + 1, by: 2) {
    for _ in 0..<spacesBefore {
      ret += " "
    }
    for _ in 0..<n {
      ret += "*"
    }
    spacesBefore -= 1
    ret += "\n"
  }
  spacesBefore = 1
  for n in stride(from: size - 2, to: 0, by: -2) {
    for _ in 0..<spacesBefore {
      ret += " "
    }
    for _ in 0..<n {
      ret += "*"
    }
    spacesBefore += 1
    ret += "\n"
  }
  return Optional(ret)
}

File: 6-kyu/give-me-a-diamond/javascript/solution.js
const diamond = (n) => {
  if (n % 2 === 0 || n < 1) {
    return null;
  }
  if (n === 1) {
    return "*\n";
  }
  let ret = [];
  let spacesBefore = Math.floor(n / 2);
  let asterisks = 1;
  for (let i = 0; i < n; i++) {
    let line = "";
    if (i < (n - 1) / 2) {
      line = " ".repeat(spacesBefore) + "*".repeat(asterisks);
      ret.push(line);
      spacesBefore--;
      asterisks += 2;
    } else if (i === (n - 1) / 2) {
      line = "*".repeat(n);
      ret.push(line);
      spacesBefore = 1;
      asterisks -= 2;
    } else {
      line = " ".repeat(spacesBefore) + "*".repeat(asterisks);
      ret.push(line);
      spacesBefore++;
      asterisks -= 2;
    }
  }
  return ret.join("\n") + "\n";
}
